sky/execution.py

```diff
@@ -633,6 +633,18 @@ def spot_launch(
         controller_task = task_lib.Task.from_yaml(yaml_path)
         controller_task.spot_task = task
         assert len(controller_task.resources) == 1
+
+        # Override the spot controller cloud if the user has specified it.
+        controller_cloud_str = os.getenv(
+            env_options.SPOT_CONTROLLER_CLOUD_ENV_VAR)
+        if controller_cloud_str is not None:
+            controller_cloud = clouds.CLOUD_REGISTRY.from_str(
+                controller_cloud_str)
+            logger.info(f'Override spot controller cloud to {controller_cloud}')
+            controller_resources = list(controller_task.resources)[0]
+            controller_task.set_resources(
+                {controller_resources.copy(cloud=controller_cloud)})
```

Comment:
Some questions:

- What's the purpose of this for testing? Is it to avoid the tests defaulting to AWS which has the cheapest default resource?
- What if the controller exists on cloud A, and this overrides it to cloud B? Would that throw an error when launching?

I kind of feel it's ok to manually launch a GCP controller so that we can avoid handling this env var.

Reply:
> What's the purpose of this for testing? Is it to avoid the tests defaulting to AWS which has the cheapest default resource?

Yes, it is to avoid using AWS for spot controller by default due to the lowest price.

> What if the controller exists on cloud A, and this overrides it to cloud B? Would that throw an error when launching?

Yes, this will throw an error if the cloud does not match, just as what it will show for `sky launch -c existing-cluster --cloud <cloud-not-match>`.

> I kind of feel it's ok to manually launch a GCP controller so that we can avoid handling this env var.

This env var is not only for the test purpose, but allowing advanced user to specify the cloud for the controller VM. I also planned to support other env vars for changing the number of CPUs of the controller as well as the number of CPUs a controller process should occupy. Wdyt?

Reply:
After discussion offline, we decided to revert the spot controller setting env var. We can have another PR for adding the option to `~/.sky/config.yaml`.